packages/client/window/src/windows/NewTab.ts

```diff
@@ -0,0 +1,65 @@
+import { urlToOrigin } from "@/utils/urlToOrigin";
+
+import type { EventMap } from "../EventEmitter";
+import type { EventEmitterWithHandshakeOptions } from "../handshake";
+import { HandshakeParent } from "../handshake/Parent";
+
+export interface NewTabWindowOptions {
+    awaitToLoad?: boolean;
+}
+
+export class NewTabWindow<IncomingEvents extends EventMap, OutgoingEvents extends EventMap> extends HandshakeParent<
+    IncomingEvents,
+    OutgoingEvents
+> {
+    private constructor(
+        public window: Window,
+        targetOrigin: string,
+        options?: EventEmitterWithHandshakeOptions<IncomingEvents, OutgoingEvents>
+    ) {
+        super(window, targetOrigin, options);
+    }
+
+    static async init<IncomingEvents extends EventMap, OutgoingEvents extends EventMap>(
+        url: string,
+        options: NewTabWindowOptions & EventEmitterWithHandshakeOptions<IncomingEvents, OutgoingEvents>
+    ) {
+        const popup = await createNewTab(url, options);
+        return new NewTabWindow<IncomingEvents, OutgoingEvents>(
+            popup,
+            options?.targetOrigin || urlToOrigin(url),
+            options
+        );
+    }
+
+    static initSync<IncomingEvents extends EventMap, OutgoingEvents extends EventMap>(
+        url: string,
+        options: EventEmitterWithHandshakeOptions<IncomingEvents, OutgoingEvents>
+    ) {
+        const popup = createNewTabSync(url);
+        return new NewTabWindow<IncomingEvents, OutgoingEvents>(
+            popup,
+            options.targetOrigin || urlToOrigin(url),
+            options
+        );
+    }
+}
+
+function createNewTabSync(url: string) {
```

Comment:
could this be private methods instead of just functions?

Reply:
ehh i dont think it really makes a difference, and this is the pattern for all of them...